File: practiques/p4_brachistochrone/brachistochrone.m
clear all;
format long;

global a
global b
a = 1/2;
b = 1;

%y = r(t - sin(t))
%x = r(1 - cos(t))

fun = @freturn;

x0 = [1 1];
x = fsolve(fun, x0)

File: practiques/p4_brachistochrone/brachistochrone.cc
#include <iostream>
#include <cmath>
using namespace std;

int main() {
  double r;
  double tf;
  cin >> r >> tf;

  int n = 5000;
  
  for (int i = 0; i <= n; ++i) {
    double t = i*tf/n;
    double x = r*(t - sin(t));
    double y = r*(1 - cos(t));
    cout << x << " " << y << endl;
  }
}
